### DISCUSSION TITLE
V5 Cart Reader not recognizing all WonderSwan carts

### DISCUSSION THREAD

**Discussion by Weef-Bellington** [sanni/cartreader#736] [Need help]
I built a WonderSwan adapter using the Save the Hero Builders board and a cartridge slot from an original WonderSwan. I've only had intermittent success with dumping carts. It dumps most WonderSwan cartridges just fine, and most of the ones that didn't work at first worked just fine after I cleaned them. 

However, it doesn't seem to recognize most WonderSwan Color carts. When I insert one, the OSCR's LCD is stuck on the word "Initializing" forever. This is true of every WonderSwan Color cart I own except for Koibito Kun. (It recognized Terrors 2, but I got a checksum error every time I tried to dump it.) It's also true of every Square game I have, including Chocobo's Mysterious Dungeon, an original WonderSwan cart.

Any idea what's going on here? Is it possible it's a problem with my soldering? I just soldered each pin on the cartridge slot to the pad underneath it. I also attached the two tantalum capacitors as recommended.

Edited to add: In case it helps, here's a bad dump of Terrors 2. 

[BANC04.zip](https://github.com/sanni/cartreader/files/10951292/BANC04.zip)

Edited to add 2: I used my multimeter to make sure all the pins in the WS connector have good connectivity to the corresponding pins on the SNES end of the board. (Pins 1 and 48 don't seem to go anywhere.)

**Comment by PsyK0p4T** [comment]
Hey :)

Problem 1 : The "Initializing..." wording appears at very startup step. It stays displayed until the MMC access is unlocked for the cart (needed for bank switching etc). If everything goes well, you don't even see it. But if the process takes too long, you'll keep it on screen forever. If it's visible for more than 5 seconds, then there is a problem and you can poweroff the reader. Most of time, it's coming from the cartridge pins which are not clean (or really damaged cart, but that's pretty rare). Best thing to do here is to open the cart up, get the pcb, and clean the pins perfectly, again and again. With a soft rubber for example.

Problem 2 : checksum error. This is really not linked to the fact that your carts are WS color. I've dumped different ones, and both mono or color ones are accepted. If the checksums are not matching, it's coming from a bad dump. And many reasons are possible : 
- write problems on your SD cart > do you have no problem with other system dumps ?
- difficulties to read your cartridge > Happens to me a lot. Sometimes it takes me 10 attempts before getting the expected crc. Eject/re-insert cart, clean a bit, retry... 

If you dumped some carts fine already, then your solderings should be fine. Double check that there is no contact between two close pins, and you should be good.
Also, I can confirm the chocobo game is dumping fine, I did it earlier.

**Reply by Weef-Bellington** [reply]
Thanks very much for your response!

Problem 1: To be clear, this problem—sticking at the "Initializing..." screen indefinitely—happens to every Color cart I try except Koibito Kun and Terrors 2. (I've only had a persistent checksum error with two carts.) Every WS cart except the two Chocobo games is recognized, and every Color cart except the two I mentioned sticks at "Initializing..."

I tried cleaning three of the problem carts very thoroughly (Chocobo's Mysterious Dungeon, Final Fantasy II, and Rhyme Rider Kerorican). I pulled out the PCB, scrubbed the pins down with a gum eraser, and wiped them with IPA until the swabs were clean. It made no difference. It seems like too big a coincidence to be just a problem with dirty carts.

Is there any reason to think bridged pins could cause this? Some of my joins have some big solder blobs on them that look very close together. I'll probably try redoing those.

Problem 2: After multiple cleanings and reseatings, I got a good checksum on Terrors 2! 

Edit: Forgot to answer one of your questions: the dumper works just fine on other systems. I've tested every slot but SMS (don't have any carts) and they all work great. Doesn't seem like an SD card problem.

**Reply by PsyK0p4T** [reply]
Ok so if you obtain good roms/crc with other consoles, it should not be related to the SD or write errors.
Here I've dumped different Color games, sometimes easily, sometimes not. Sometimes with the good crc directly, and sometimes not. But at the really end all the ones I've tried ended to work (exept one, but I suspect it to be physically damaged).
More things you can check :
- open up the cart and confirm that there is no rust or corrosion. I had some carts like this, and this caused bridges and shortcuts between chip pins.
- make sure you're using 3V on power switch
- use a multimeter to check your solders on the adapter, to confirm that two close pins are not bridged
- can you list the games which are stuck on the init message ? I will compare with mines to see if I dumped some of them successfully

**Reply by Weef-Bellington** [reply]
Thanks again for your help and patience!

- I visually confirmed no corrosion on the PCB for any of the carts that are giving me trouble.
- I was on the 5V switch for most of my tests, so thanks for pointing that out! I kept it on 3V today, and the problem persists.
- I tested the pins with my multimeter. Pins 24 and 25 are bridged. Could this be the problem? I did an ugly job installing the capacitors, so maybe I accidentally bridged the two pads underneath one of the caps with solder.

I'm still stuck on the init screen for these games as of tonight. There are still six more to test.

- S.D. Gundam Operation U.C.
- Chocobo's Mysterious Dungeon
- Hataraku Chocobo
- Final Fantasy II
- Final Fantasy IV
- Front Mission
- Meitantei Conan: Yuugure no Oujo
- Rhyme Rider Kerorican
- Romancing Saga
- Wild Card

**Reply by splash5** [reply]
Pin 24 and 25 are both 3V power so that is ok. Pin 1 and 48 are go to GND.
Stuck in "Initializing..." screen with a WSC game means that MMC can not be initialized or the header is bad.
You can also check if pin 47 is soldered without problem.

Use an eraser (for pencil) or IPA to clean cartridge's gold finger even if they looks clean. Also make sure that adapter PCB's gold finger is clean.
I can confirm the games you listed except "S.D. Gundam Operation U.C." and "Front Mission" can be dumped without any problem.

And.. what kind of adapter you use? V1 or V2?

**Reply by Weef-Bellington** [reply]
Thanks, splash5! I've cleaned every cart I've tried as thoroughly as I know how. I gave the adapter pins a clean just in case, with no difference.

I get good continuity from pin 47 to pin 1 on the adapter.

I bought this adapter board from Save the Hero Builders: https://savethehero.builders/collections/build-service/products/ws-adapter-only-pcb-board
I don't know what version it is.

**Reply by splash5** [reply]
Ok, so the adapter PCB you use is a V1, which is NOT compatible with cartreader V4 (and V5).
I have designed a V2 adapter PCB which supports cartreader V3 and later revision.
For V2 adapter PCB, you can check here #337.

If you want to keep using the WS adapter you bought, you can do a small modification (cut one trace and solder one wire) to make your WS adapter to support cartreader V5. For more details please check [here](https://twitter.com/wxhsplash/status/1496753353124036610?s=20). (a twitter link)

**Reply by Weef-Bellington** [reply]
Thanks for pointing that out!! I worried I was losing my mind. 

I may try the modifications you suggest, though with my version of the board it looks like I would have to attach the wire directly to pin 47 since all the traces are coated with something. Would I also need to alter the code? Your tweet mentions changing a few lines.

I've never bought a custom PCB before, but I'll figure out how to get myself a copy of your v2 adapter.  Thanks again!

**Reply by splash5** [reply]
You can use any method you prefer to do the adapter modification, just cut the trace originally connect between SNES side pin 1 and WS side pin 47, then connect SNES side pin 33 to WS side pin 47.

And you don't need to modify any code, all the code need to be changed has been merged into project, so only make sure you select right hardware revision when compling/updating cartreader software and everything should work.

**Reply by Weef-Bellington** [reply]
Great! I'll give it a try as soon as I can get my hands on some wire and then report back.

**Reply by Weef-Bellington** [reply]
The modifications did the trick! I was able to dump viable ROMs of all the carts I own, including the ones I listed as giving me trouble. 

Thanks so much to both PsyK0p4T and splash5 for helping me troubleshoot this issue.
